docs/azure/sdk/snippets/dependency-injection/HostBuilder/Program.cs

```diff
@@ -1,20 +1,48 @@
-﻿using HostBuilder;
-#region snippet_HostBuilder
+﻿#region snippet_HostBuilder
 using Azure.Identity;
+using Azure.Messaging.ServiceBus;
+using Azure.Messaging.ServiceBus.Administration;
 using Microsoft.Extensions.Azure;
 
+List<string> queueNames = await GetQueueNames();
+
 IHost host = Host.CreateDefaultBuilder(args)
     .ConfigureServices(services =>
     {
-        services.AddHostedService<Worker>();
         services.AddAzureClients(clientBuilder =>
         {
+            // Register clients for each service
             clientBuilder.AddSecretClient(new Uri("<key_vault_url>"));
             clientBuilder.AddBlobServiceClient(new Uri("<storage_url>"));
+            clientBuilder.AddServiceBusClientWithNamespace("<your_namespace>.servicebus.windows.net");
             clientBuilder.UseCredential(new DefaultAzureCredential());
+
+            // Register a subclient for each Service Bus Queue
+            foreach (string queue in queueNames)
+            {
+                clientBuilder.AddClient<ServiceBusSender, ServiceBusClientOptions>((_, _, provider) =>
+                    provider.GetService<ServiceBusClient>().CreateSender(queue)
+                ).WithName(queue);
+            }
         });
-    })
-    .Build();
+    }).Build();
 
 await host.RunAsync();
+
+async Task<List<string>> GetQueueNames()
+{
+    // Query the available queues for the Service Bus namespace.
+    var adminClient = new ServiceBusAdministrationClient
+        ("<your_namespace>.servicebus.windows.net", new DefaultAzureCredential());
+    var queueNames = new List<string>();
+
+    // Because the result is async, the queue names need to be captured to a standard list to avoid async
+    // calls when registering. Failure to do so results in an error with the services collection.
```

Comment:
```suggestion
    // Because the result is async, the queue names need to be captured
    // to a standard list to avoid async calls when registering. Failure to
    // do so results in an error with the services collection.
```